src/service/standard-actions-impl.js

```diff
@@ -125,6 +131,21 @@ export class StandardActions {
    * @param {!./action-impl.ActionInvocation} invocation
    * @private
    */
+  handleAmpNavigateTo_(invocation) {
+    if (!invocation.satisfiesTrust(ActionTrust.HIGH)) {
+      return;
+    }
+    const url = invocation.args['url'];
```

Comment:
use `isProtocolValid` from `url.js` to assert that url is not dangerous (also a test for `navigateTo(javascript:)` to ensure the validation.